tests/random_test.py

```diff
@@ -394,6 +394,27 @@ def testGammaGradType(self):
     # Should not crash with a type error.
     api.vjp(f, a, b)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_lam={}_{}".format(lam, dtype),
+       "lam": lam, "dtype": onp.dtype(dtype).name}
+      for lam in [0.5, 5, 50, 500]
+      for dtype in [onp.int32, onp.int64]))
+  def testPoisson(self, lam, dtype):
```

Comment:
Maybe test batches? This is mainly to check that rejection works correctly over the batch members (since the batch members are "tied" together by the reject steps).

Reply:
Arguably everything we're doing here is batched, since we're generating 10000 samples at once. Though I agree that it would be good to verify that things work properly when different elements of the batch are generated with different rates.